### PR TITLE
utility function to get module name

### PR DESCRIPTION
**Pull request by fabianburth**
## Description
This Pull Request provides a utility function to get the module's name.

### CODE REVIEW

File: pkg/utils/package.go [hunk 1]
```diff
@@ -2,12 +2,56 @@ package utils
...
+	"github.com/mandelsoft/vfs/pkg/osfs"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"golang.org/x/mod/modfile"
 	"reflect"
 	"runtime"
 	"strings"
 )
 
-const MODULE_PATH = "github.com/open-component-model/ocm"
+func GetRelativePathToProjectRoot(i ...int) (string, error) {
```
Comment (hilmarf) [inline pkg/utils/package.go:15]: may I ask for golang-doc for that function, please?
What's the optional parameter 'i' for?

File: pkg/utils/package.go [hunk 2]
```diff
@@ -2,12 +2,56 @@ package utils
...
+	}
+	return "", nil
+}
+
+func GetModuleName() (string, error) {
+	pathToRoot, err := GetRelativePathToProjectRoot()
+	if err != nil {
+		return "", err
+	}
+	pathToGoMod := filepath.Join(pathToRoot, "go.mod")
```
Comment (hilmarf) [inline pkg/utils/package.go:39]: would it make sense to spend a public const for: go.mod?

```go
const GoMod = "go.mod"
```

**Review by hilmarf — CHANGES_REQUESTED** [review]
Please document exported functions with go-doc

File: pkg/utils/package.go (hunk 1 above)
Comment (fabianburth) [reply pkg/utils/package.go:22]: added documentation

File: pkg/utils/package.go (hunk 2 above)
Comment (fabianburth) [reply pkg/utils/package.go]: sure, why not

**Comment by fabianburth** [comment]
> Please document exported functions with go-doc

resolved the requested changes

**Review by mandelsoft — DISMISSED** [review]

File: pkg/utils/package.go [hunk 3]
```diff
@@ -24,21 +74,59 @@ func GetPackageNameForFunc(i interface{}) (string, error) {
...
+	funcIndex := strings.Index(fullFuncName[lastSlashIndex:], ".")
+	packagePath := fullFuncName[:lastSlashIndex+funcIndex]
+
 	return packagePath, nil
 }
 
+// GetPackageNameForCaller returns the package name of the function calling this function.
+// The parameter offset is useful if this function is used within another utility function. With an offset of 1, you
+// could specify that it should not return the package name of the immediate caller (the other utility function), but
+// the package name of the function calling the utility function.
```
Comment (hilmarf) [inline pkg/utils/package.go:89]: 👍

**Review by hilmarf — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Pull request merged**